### PR TITLE
WIP: Better warning when comma is used instead of semicolon in record

### LINKED ISSUE #1122: Improve error reporting: Accidental use of , instead of ; as record field separator
## What
The following error message is generated from this code snippet:
```fsharp
type Person = { Name : string; Age : int; City : string }
let x = { Name = "Isaac", Age = 21, City = "London" } // , instead of ;
```

``error FS0001: This expression was expected to have type
    string    
but here has type
    'a * 'b * 'c``

## Why
Newcomers to F# syntax (or otherwise!) may accidentally use ``,`` instead of ``;`` as field separators. The error message is not helpful in this case - the inclusion of generics in the message further complicates matters and does not aid the developer with the real problem.

## How
Replace the error message with something like: -

``F# uses ; to separate fields values when creating a record, not , e.g. let x = { Name = "Isaac"; Age = 21 ... }``

### PR DESCRIPTION
**Pull request by forki**
fixes #1122

image

### CODE REVIEW

**Comment by forki** [comment]
chenged "Consider to replace" to "Consider replacing" as @isaacabraham suggested

**Comment by smoothdeveloper** [comment]
I was noticed of a bikeshedding type of job from twitter.

Can we not mention `F#` in the error message?

`';' is used to separate field values in records. Consider replacing ',' with ';'.`

Great job on the fix nonetheless :)

**Comment by forki** [comment]
There are other error messages that contain the string "F#". Maybe create a
new issue which discusses to make that consistent. Also it would make a
really nice beginner friendly issue to fix.

**Comment by smoothdeveloper** [comment]
> really nice beginner friendly issue to fix

Yes way more than the one with suggesting inline, I looked at it for a while and not getting close to a resolve on it at this point.

I'll make a ticket to remove the redundant mention of F# in the messages, I think we shouldn't add more though :smile:

**Comment by vilinski** [comment]
is it possible to do the similar with lists? happens me often when switching from other languages

**Comment by forki** [comment]
That one is not so easy. See https://github.com/Microsoft/visualfsharp/issues/1120

**Comment by forki** [comment]
I removed "F#" from the message.

File: src/fsharp/ConstraintSolver.fsi [hunk 1]
```diff
@@ -56,6 +56,10 @@ type ContextInfo =
 | OmittedElseBranch
 /// The type equation comes from an else branch that might have different type than the corresponding if branch.
 | ElseBranchWithDifferentType
```
Comment (dsyme) [inline src/fsharp/ConstraintSolver.fsi]: Off topic, but it feels like this context should just be called ``ElseBranch``

**Comment by dsyme** [comment]
This looks good to me (I haven't checked out why AppVeyor CI is failing).  Great to see that the "context" is consistently helping us improve error messages.

**Comment by dsyme** [comment]
Re ``F#`` in error messages - I did scan these the other day, and am happy that this isn't chronic - most uses are specific cases where we decided to refer to the language e.g. w.r.t. some OCaml compat feature.    So I don't think we need to do anything there.

**Comment by KevinRansom** [comment]
@dsyme ,  @ forki ,

Appveyor is failing because of the test baseline needs updating with the revised error text.

I will pull this and fix the test at the same time.
````
[00:35:57] +++ Warnings (CommaInRecCtor.fs) +++
[00:35:57] Microsoft (R) F# Compiler version (private)
[00:35:57] Copyright (c) Microsoft Corporation. All Rights Reserved.
[00:35:57] (5,18): error FS0001: This expression was expected to have type
[00:35:57]     string    
[00:35:57] but here has type
[00:35:57]     'a * 'b * 'c    
[00:35:57] 
[00:35:57] A ';' is used to separate field values in records. Consider replacing ',' with ';'.
[00:35:57] (5,9): error FS0764: No assignment given for field 'Age' of type 'CommaInRecCtor.Person'
[00:35:57] 
[00:35:57] *** The following necessary lines were never matched:
[00:35:57] ***	\(5,18\):.+error FS0001:.+Consider replacing
[00:35:57] 
[00:35:57] 
[00:35:57] *** The following necessary lines were incorrectly matched:
[00:35:57] 
[00:35:57] Unexpected Compiler Output 
[00:35:57] FAIL
[00:35:57] 
[00:35:57] 
[00:35:57] 
````

**Comment by forki** [comment]
this was merged by @KevinRansom 2s before I pushed small changes. will reopen new PR

**Pull request closed without merging**